grype/matcher/common/distro_matchers.go

```diff
@@ -15,7 +15,7 @@ func FindMatchesByPackageDistro(store vulnerability.ProviderByDistro, d *distro.
 		return nil, nil
 	}
 
-	verObj, err := version.NewVersionFromPkg(p)
+	verObj, err := version.NewVersionFromPkg(p) // Incorrect version if desynced
```

Comment:
I'm not certain I follow this comment (I think this is listed in your PR description as a question to ask too)

Reply:
Yea I'm removing it now - I was just a placeholder so I could remember where the incorrect version would be placed when direct and indirect were not the same version.